### ISSUE TITLE
Source code without comments

### ISSUE THREAD

**Issue by dungwinux** [minhducsun2002/codefun.vn#1]
- Add comments so readers could understand what they are doing.

**Comment by minhducsun2002** [comment]
Will add periodically, based on the number of solutions posted.
Currently, comments will be added after solution for P00020 posted.

**Comment by dungwinux** [comment]
- What about `#include <bits/stdc++.h>` ?

**Comment by minhducsun2002** [comment]
In order to keep the code as short as possible, `#include <bits/stdc++.h>` will be used when two or more headers are required for a successful compilation.

**Comment by NOVAglow** [comment]
I'm not related but it's weird seeing you guys speaking English?

Also poor Dzung gets issue closed without a gentle reply.

**Comment by minhducsun2002** [comment]
@NOVAglow, I made the necessary change at most two or three hours after notification. Please, tell me what's the point of keeping it open? (`¯\_(ツ)_/¯`)
(Also, we're on [GitHub](github.com). Be __international__ :+1:)

**Comment by NOVAglow** [comment]
Keep issue open until comments are added.

**Comment by NOVAglow** [comment]
I'm not complaining about speaking English on GitHub. I just felt weird but that's about it.

**Comment by minhducsun2002** [comment]
Issue reopened.

**Comment by NOVAglow** [comment]
Just a suggestion.
```cpp
#include <iostream>
int main()
{
    int i;  //declare local variable
    std::cin >> i;  //set the variable to user's input
    std::cout << (i << 1);  //print the doubled value of the variable
                            // "<<" means bitwise shift-left, which implies a multiplication by 2.
}
```

Should have been...
```cpp
#include <iostream>
int main()
{
    int i;                  // Declare a local variable, which is an integer
    std::cin >> i;          // Set the variable to user's input
    std::cout << (i << 1);  // Print the doubled value of the variable
    // "<<" is the shift-left operator, which is a multiplication by a power of 2
    // i << 1 is equal to i * 2, i << 2 is equal to i * 4, i << 3 is equal to i * 8
}
```

for more precision. :smiley: 

Also you are that type of person who puts a single semi-colon on a single line.

**Comment by minhducsun2002** [comment]
Please explain what do you mean ? :+1: 
> that type of person who puts a single semi-colon on a single line.

Also, thank you for helping me with the indentation. With four files open at the same time, my memory has started to degrade, leading to the fact that I don't pay enough attention about readability.

**Comment by NOVAglow** [comment]
> Please explain what do you mean ?

You:
```cpp
int main()
{
}
```

Me:
```cpp
int main() {
}

**Comment by minhducsun2002** [comment]
Definitely, (from my perspective) it's better to keep those curly braces on their own line. It helps me distinguish between a function's declaration and its internal working.

However if you find 
```cpp
int main() {
}
```
easier to read and less cumbersome than
```cpp
int main()
{
}
```
then that's good :+1:

**Comment by minhducsun2002** [comment]
Also, if you have time, could you please help me with those [`explanation`](https://github.com/minhducsun2002/codefun.vn/tree/explanation)s?

**Comment by NOVAglow** [comment]
Yeah sure I'll review them.

**Comment by NOVAglow** [comment]
I think I will help when you aren't working on this repository.

**Comment by minhducsun2002** [comment]
Also, could you help me with [P00420](https://codefun.vn/problems/P00420), @dungwinux, @NOVAglow? I'm finding it difficult to write easy-to-understand solution for that problem.

**Comment by NOVAglow** [comment]
What's your explanation to it?

**Comment by NOVAglow** [comment]
By the way, can you please tell the guy who maintains codefun.vn to license the site's JavaScript code under a free software license and add license notices to the source code? I tried (i.e. attempted to) mailing Mr. Ho Dac Phuong but got no reply.

**Comment by minhducsun2002** [comment]
@NOVAglow Sorry, I should have created a separate link to my solution.

**Comment by NOVAglow** [comment]
I supposed `return ((i % 10 ) + '0'` doesn't really convert the exact integer to character?

**Comment by minhducsun2002** [comment]
Also, if you can't contact Mr. Ho Dac Phuong, you may consider contacting @natsukagami.

**Comment by minhducsun2002** [comment]
Yes. That function will be there until I can deal with input sanitization.

**Comment by NOVAglow** [comment]
Do I have to add explanation to `to_char()` and `to_digit()`? I so I have to add a...considerable amount of lines of code to the source.

**Comment by minhducsun2002** [comment]
Oh, sure. `add` is addition, `mul` is multiplication. For unsigned integers.

**Comment by NOVAglow** [comment]
Sorry. I mean the programmer-defined `add` and `mul` in P00420.

**Comment by minhducsun2002** [comment]
Yes, it is. I `Ctrl+C`ed and `Ctrl+V`ed from [`bigint.cpp`](https://github.com/minhducsun2002/Algorithms-Implementations/blob/master/Big%20Integers/bigint.cpp).

**Comment by NOVAglow** [comment]
Apparently I'm having troubles trying to summarize these functions. What is P00420 trying to deal?

**Comment by NOVAglow** [comment]
I'd rather you documented your code with more details.

**Comment by minhducsun2002** [comment]
I'll try simplify the problem.

Given two numbers written on the same line, find their product.
A input of
```
2 5
``` 
will yield a result of
```
10
```

Those numbers are unsigned integers. They're capped at `1e100`.

**Comment by minhducsun2002** [comment]
Sure. Originally, some lines of code could have been more descriptive. But in order to keep it easier to read, I started to shorten them with ternary operator.

Anyway, the commit history have some traces of the longer code.

**Comment by NOVAglow** [comment]
I just managed to install G++.

And nah, those are strings, converted to integers. I think you've over-complicated the additional and multiplication process.

**Comment by minhducsun2002** [comment]
But the `1e100` limit, as you know, greatly exceeds the limits of `long long int` in C++. Using `long double` will make the output process increasingly more complicated. Since I've written code for the addition and multiplication operator, I think it's better to use them. Anyway, it's been tested.

**Comment by minhducsun2002** [comment]
You are encouraged to simplify it anyway :+1:.

**Comment by NOVAglow** [comment]
I'm not encouraged to simplify it. I just see code and want to add a little bit to it.

**Comment by minhducsun2002** [comment]
Well, you're welcome :+1:.

**Comment by dungwinux** [comment]
It's better to write short problems details in header of source code

**Comment by NOVAglow** [comment]
Someone is not documenting his code. :/

**Comment by minhducsun2002** [comment]
@NOVAglow: I think that if you still wish to have [`codefun.vn`](https://codefun.vn)'s source code, the only way that works now (as far as I know) is to fetch all resources. I don't know much about JavaScript, so I cannot verify if what I read through Chromium's Developer Tools has been obfuscated. Give it a try if you decide to.

@natsukagami: Dạ, chúng em cảm ơn câu trả lời của anh ạ.

**Comment by NOVAglow** [comment]
I don't wish to have codefun.vn's source code. No I don't. I was just asking if the maintainer could liberate the JavaScript code on that site. It's not too much of a deal. Instead of adding license notices then you can just have magnet link. Just about 3 more lines of comments. Not too much of a problem.

However, codefun.vn is not open source, and if there's really a sincere reason for not open-sourcing it, then there's no use of liberating it anyway.

If you liberate JavaScript, you lose nothing. In fact you expand to more audiences.

**Comment by natsukagami** [comment]
Yes there's a strong reason not to open source it: I don't want to. And you cannot do anything about it.
I don't care about people who are trying to read my minified code. It was built out of a few if-else and a mix of open source libraries anyway. There isn't much intelligence there. I don't have the interest of having any audiences contributing on the website, nor I want people to come and read my code either. I have a team maintaining the thing and in **my** humble opinion, in this case, that's better than people I don't know.

**Comment by minhducsun2002** [comment]
@NOVAglow : I'm not familiar with JavaScript. May you please give me an explanation on what's really called "minification"? Having been an user of Chromium, I sometimes encounter this : 

![image](https://user-images.githubusercontent.com/32769878/40278496-1410fd7c-5c5c-11e8-9fdb-a394d298b60b.png)

in its Developer Tools panel.
And even though I do understand what `pretty-print` mean, I still don't understand what `minified file` mean.

**Comment by Banbeucmas** [comment]
Urgh, chúng ta có thể kết thúc việc này đc chưa.

1. Đ cần phải be International ở đây vì cả tá bài Tiếng Việt có đề Tiếng Việt mà phải cmt Tiếng Anh cả. Vãi cả quan hệ mẫu thân nhà nó, thằng này từng qua 1 cái Repo mà cái issue toàn Tiếng Pháp ko. Vì sao, vì cái Project chết tiệt đó chỉ định làm riêng cho người Pháp. Làm ơn, đừng để cái độ cringe nó vãi cả háng đến mức thằng này đang thiếu khí điều hòa rồi chỉ muốn bay lên trời cho lành.
2. Hỏi xin Src codefun. Đừng đùa thằng này... Không phải cái khỉ gì cũng Open Source đc. Điều này giống hệt việc mấy chú bảo Admin CodeForce : "Ê các chú open source trang web đc ko?" Câu tl là khống.

Cai Repo này đăng bài giải đã là 1 chuyện rồi. Nhưng đừng biến nó thành cringyzone ko thì thằng này đến những đứa khác nhìn xong đến mức tự sọc tiết chính mình mất.

**Comment by NOVAglow** [comment]
I repeat that I do **not** ask for codefun.vn source code, so you don't have to show me ways of viewing the source. Read the comment again if you may.

We are supposed to be discussing about code not having explanations in this issue page, and I already ended the thread about codefun.vn JavaScript stuff, so no more about it. Let's just get things done.
